File: BotsSDK/korebotsdklib/src/main/java/kore/botssdk/models/BrandingBodyAgentMsgModel.java
package kore.botssdk.models;

import java.io.Serializable;

import kore.botssdk.utils.StringUtils;

public class BrandingBodyAgentMsgModel implements Serializable {
    private String bg_color;
    private String color;
    private String separator;
    private BrandingIconModel icon;
    private BrandingTitleModel title;
    private BrandingTitleModel sub_title;

    public BrandingBodyAgentMsgModel updateWith(BrandingBodyAgentMsgModel configModel) {
        bg_color = !StringUtils.isNullOrEmpty(configModel.bg_color) ? configModel.bg_color : bg_color;
        color = !StringUtils.isNullOrEmpty(configModel.color) ? configModel.color : color;
        separator = !StringUtils.isNullOrEmpty(configModel.separator) ? configModel.separator : separator;
        icon = icon != null && configModel.icon != null ? icon.updateWith(configModel.icon) : icon;
        title = title != null && configModel.title != null ? title.updateWith(configModel.title) : title;
        sub_title = sub_title != null && configModel.sub_title != null ? sub_title.updateWith(configModel.sub_title) : sub_title;
        return this;
    }

    public void setBg_color(String bg_color) {
        this.bg_color = bg_color;
    }

    public void setColor(String color) {
        this.color = color;
    }

    public void setIcon(BrandingIconModel icon) {
        this.icon = icon;
    }

    public void setTitle(BrandingTitleModel title) {
        this.title = title;
    }

    public void setSub_title(BrandingTitleModel sub_title) {
        this.sub_title = sub_title;
    }

    public void setSeparator(String separator) {
        this.separator = separator;
    }

    public String getBg_color() {
        return bg_color;
    }

    public BrandingTitleModel getTitle() {
        return title;
    }

    public BrandingTitleModel getSub_title() {
        return sub_title;
    }

    public BrandingIconModel getIcon() {
        return icon;
    }

    public String getColor() {
        return color;
    }

    public String getSeparator() {
        return separator;
    }
}

File: BotSdk/botclient/src/main/java/com/kore/network/api/responsemodels/branding/BrandingBodyAgentMsgModel.kt
package com.kore.network.api.responsemodels.branding

import com.google.gson.annotations.SerializedName

data class BrandingBodyAgentMsgModel(
    @SerializedName("bg_color")
    val bgColor: String = "",
    @SerializedName("color")
    val color: String = "",
    @SerializedName("separator")
    val separator: String = "",
    @SerializedName("icon")
    val icon: BrandingIconModel,
    @SerializedName("title")
    val title: BrandingTitleModel,
    @SerializedName("sub_title")
    val subTitle: BrandingTitleModel,
)

fun BrandingBodyAgentMsgModel.updateWith(configModel: BrandingBodyAgentMsgModel): BrandingBodyAgentMsgModel {
    return this.copy(
        bgColor = configModel.bgColor.ifEmpty { this.bgColor },
        separator = configModel.separator.ifEmpty { this.separator },
        color = configModel.color.ifEmpty { this.color },
        icon = icon.updateWith(configModel.icon),
        title = title.updateWith(configModel.title),
        subTitle = subTitle.updateWith(configModel.subTitle)
    )
}